xclim/core/units.py

```diff
@@ -708,3 +710,20 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return dec
+
+
+def check_same_units_and_convert(func) -> Callable:
+    def _check_same_units(sim, ref, **kwargs):
+        units_sim = units2pint(sim.units)
+        units_ref = units2pint(ref.units)
+
+        if units_sim != units_ref:
+            warn(
+                f" sim({units_sim}) and ref({units_ref}) don't have the same units."
+                f" sim will be converted to {units_ref}."
+            )
+            sim = convert_units_to(sim, ref)
+        out = func(sim, ref)
+        return out
+
+    return _check_same_units
```

Comment:
If the function is specific to sdba (which it is because of the input names), I would leave it in sdba. This can go directly in `sdba.measures`.

Reply:
done in the latest commit